src/lib/utils/ghash/ghash.cpp

```diff
@@ -60,16 +58,16 @@ void GHASH::ghash_multiply(secure_vector<uint8_t>& x, std::span<const uint8_t> i
       uint64_t Z[2] = {0, 0};
 
       for(size_t i = 0; i != 64; ++i) {
-         const uint64_t X0MASK = (ALL_BITS + (X[0] >> 63)) ^ ALL_BITS;
-         const uint64_t X1MASK = (ALL_BITS + (X[1] >> 63)) ^ ALL_BITS;
+         const auto X0MASK = CT::Mask<uint64_t>::expand_top_bit(X[0]);
+         const auto X1MASK = CT::Mask<uint64_t>::expand_top_bit(X[1]);
```

Comment:
This area was not flagged in the paper but seems an obvious target for Clang's range analysis